### PR TITLE
fix(igx-list): Creating an index export file for the list #1939

### LINKED ISSUE #1939: IgxListItemComponent and the two template directives are missing from public_api.ts.
### Description  
IgxListItemComponent, IgxEmptyListTemplateDirective and IgxDataLoadingTemplateDirective are not listed in public_api.ts.

 * igniteui-angular version: 6.1.x
 * browser: N/A

### Steps to reproduce  

1. Try to import IgxListItemComponent, IgxEmptyListTemplateDirective and/or IgxDataLoadingTemplateDirective from public_api.

### Result  
Those items are missing.

### Expected result  
They should be listed in public_api.

### PR DESCRIPTION
**Pull request by kdinev**
Closes #1939

### CODE REVIEW

File: src/app/drop-down/drop-down.sample.ts [hunk 1]
```diff
@@ -123,9 +123,9 @@ export class DropDownSampleComponent implements OnInit {
         this.igxDropDown.toggle(overlaySettings);
     }
 
-    onSelection() {
+    onSelection(event) {
```
Comment (gedinakova) [inline src/app/drop-down/drop-down.sample.ts]: Is this change intentional?

Comment (kdinev) [reply src/app/drop-down/drop-down.sample.ts]: Yes, the demos weren't building with --prod --aot=true. This is just in the demos.
